kafka/producer/record_accumulator.py

```diff
@@ -174,28 +176,41 @@ class RecordAccumulator(object):
         'metric_group_prefix': 'producer-metrics',
     }
 
+    _COMPRESSORS = {
+        'gzip': LegacyRecordBatchBuilder.CODEC_GZIP,
+        'snappy': LegacyRecordBatchBuilder.CODEC_SNAPPY,
+        'lz4': LegacyRecordBatchBuilder.CODEC_LZ4,
+        None: LegacyRecordBatchBuilder.CODEC_NONE
+    }
+
     def __init__(self, **configs):
         self.config = copy.copy(self.DEFAULT_CONFIG)
         for key in self.config:
             if key in configs:
                 self.config[key] = configs.pop(key)
 
+        # Convert compression type to INT presentation. Mostly for unit tests,
+        # as Producer should pass already converted values.
+        ct = self.config["compression_type"]
+        self.config["compression_type"] = self._COMPRESSORS.get(ct, ct)
```

Comment:
It's a little confusing to reuse the same configuration name here. It means we have to remember that when talking to users they need this to be a string, but when talking to developers / writing tests / etc, they may need this to be an int. Can we use an internal config like `_compression_type_int` or `compression_type_int` ?